packages/roosterjs-editor-core/lib/corePlugins/DOMEventPlugin.ts

```diff
@@ -233,11 +233,15 @@ export default class DOMEventPlugin implements PluginWithState<DOMEventPluginSta
         const allItems: any[] = [];
         const searcher = this.editor?.getContentSearcherOfCursor();
         const elementBeforeCursor = searcher?.getInlineElementBefore();
+        const selection = this.editor?.getSelectionRangeEx();
 
         let eventTargetNode = event.target as Node;
         if (event.button != 2 && elementBeforeCursor) {
             eventTargetNode = elementBeforeCursor.getContainerNode();
         }
+        if (event.button == 2 && selection?.type === SelectionRangeTypes.ImageSelection) {
+            eventTargetNode = selection.image;
```

Comment:
Do we need this changes?

Reply:
Not really, but having image selection be the default provider of the node in this case makes sense. We rarely use the wrapper target on any action